manifest/provider/apply.go

```diff
@@ -564,16 +575,24 @@ func (s *RawProviderServer) ApplyResourceChange(ctx context.Context, req *tfprot
 			_, err := rs.Get(ctxDeadline, rname, metav1.GetOptions{})
 			if err != nil {
 				if apierrors.IsNotFound(err) {
-					s.logger.Trace("[ApplyResourceChange][Delete]", "Resource is deleted")
+					s.logger.Trace("[ApplyResourceChange][Delete]", "Resource is already deleted")
```

Comment:
We actually don't need the change over here. 

I this case, receiving a 404 while waiting for a resource to actually be deleted is a confirmation of the actual deletion, not an indication that the resource was not actually present to begin with (in that case the 404 will have shown up as response to the original .Delete() call).